lib/site_prism/dsl.rb

```diff
@@ -228,27 +222,35 @@ def create_error_method(name)
       end
 
       def deduce_iframe_scope_find_args(args)
+        warn_on_invalid_selector_input(args)
         case args[0]
-        when Integer
-          [args[0]]
-        when String
-          [:css, args[0]]
-        else
-          args
+        when Integer then [args[0]]
+        when String  then [:css, args[0]]
+        else args
         end
       end
 
       def deduce_iframe_element_find_args(args)
+        warn_on_invalid_selector_input(args)
         case args[0]
-        when Integer
-          "iframe:nth-of-type(#{args[0] + 1})"
-        when String
-          [:css, args[0]]
-        else
-          args
+        when Integer then "iframe:nth-of-type(#{args[0] + 1})"
+        when String  then [:css, args[0]]
+        else args
         end
       end
 
+      def warn_on_invalid_selector_input(args)
+        return if looks_like_xpath?(args[0])
+
+        msg = 'The arguments passed in look like xpath. Check your locators.'
+        SitePrism.logger.warn(msg)
+        SitePrism.logger.debug("Default locator: #{Capybara.default_selector}")
+      end
+
+      def looks_like_xpath?(arg)
+        arg.is_a?(String) && arg.start_with?('/', './')
```

Comment:
I have a feeling there is still an error here.
condition above should return if it is not-xpath-like string or another type (an integer for example)
so we assert it is a string with `arg.is_a?(String)` if it is false we should exit so let's write it down
`return if !arg.is_a?(String)`, next, if it is a string we check if it doesn't start with a slash `arg.start_with?('/', './')`. so the final condition is `return if !arg.is_a?(String) || arg.is_a?(String) && arg.start_with?('/', './')` or after simplifications `!A || A && B` => `!(A && (!A || !B))` => `!( A && !A || A && !B)` => `!(A && !B)` => `!A || B` => `return if !arg.is_a?(String) || arg.start_with?('/', './')`